src/IdentityServer4/src/Endpoints/Results/CheckSessionResult.cs

```diff
@@ -45,7 +48,18 @@ private void AddCspHeaders(HttpContext context)
         }
         private string GetHtml(string cookieName)
         {
-            return Html.Replace("{cookieName}", cookieName);
+            if (cookieName != LastCheckSessionCookieName)
```

Comment:
Maybe I overlooked something here, but having only FormattedHtml available will not allow us to check for changed cookie names, as discussed with @msschl above, ``IdentityOptions`` is not a proper singleton so someone could change the cookieName during runtime.

Reply:
I can't imagine why someone would change it dynamically. It's in DI as a singleton and intended to be consistent for the running server.

Reply:
I understand that, if you want it changed to FormattedHtml and ignore the change inconsistencies I can obviously do it, but it might later lead to bugs which are hard to find. It's your code after all.